src/ext/sse.js

```diff
@@ -196,6 +196,17 @@ This extension adds support for Server Sent Events to htmx.  See /www/extensions
 				}
 			}
 		});
+
+		var closeAttribute = api.getAttributeValue(elt, "sse-close");
```

Comment:
Using this selector is not correct. `registerSSE` is not necessarily processes elements with `sse-connect` on the root.

The following test case, for example, is failing.

```js
    it('is closed after close message from server in nested content', function() {
        var div = make('<div hx-ext="sse"><div sse-connect="/foo" sse-close="close">' +
            '<div id="d1" sse-swap="e1"></div>' +
            '</div></div>');
        this.eventSource.wasClosed().should.equal(false);
        this.eventSource.sendEvent("close");
        this.eventSource.wasClosed().should.equal(true);
    })
```

I would either move this piece of code to `ensureEventSource` or use `queryAttributeOnThisOrChildren` and then look through the hierarchy upwards for the SSE element

Reply:
I see, and thanks for the clear explanation of the problem. It does seem most appropriate inside `ensureEventSource` so I moved it to there, plus included the test you referenced above.